tapchannel/aux_closer.go

```diff
@@ -258,7 +258,9 @@ func (a *AuxChanCloser) AuxCloseOutputs(
 		// anchor amt, then we'll just drop this allocation, and modify
 		// our asset allocation to match this value.
 		if amtAfterAnchor <= o.DustLimit {
-			localAlloc.BtcAmount = btcAmt
+			if localAlloc != nil {
+				localAlloc.BtcAmount = btcAmt
+			}
```

Comment:
If `btcAmt > 0` here then shouldn't `localAlloc != nil`? In other words, do we need to create an allocation here somehow?

Reply:
No, the `localAlloc` is the allocation for assets.  And if there are more than 1000 sats in BTC available, they go into a separate, BTC only allocation (so to not mix the asset and BTC outputs too much). This `if` is for when we have neither BTC nor assets.